sable_ircd/src/server/config.rs

```diff
@@ -7,7 +10,63 @@ pub struct ListenerConfig {
     pub tls: bool,
 }
 
+#[derive(Debug, Deserialize, Clone)]
+pub struct InfoPaths {
+    pub motd: Option<PathBuf>,
+}
+
 #[derive(Debug, Deserialize, Clone)]
 pub struct ClientServerConfig {
     pub listeners: Vec<ListenerConfig>,
+    #[serde(flatten)]
+    pub info_paths: InfoPaths,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct Infos {
+    pub motd: Option<String>, // Linewise to not repeatedly split
+}
+
+impl Infos {
+    pub fn load(paths: &InfoPaths) -> Result<Infos, ConfigProcessingError> {
+        Ok(Self {
+            motd: Self::get_info(&paths.motd, "motd")?,
+        })
+    }
+
+    fn get_info(
+        path: &Option<PathBuf>,
+        name: &str,
+    ) -> Result<Option<String>, ConfigProcessingError> {
+        match path {
+            Some(real_path) => Ok(Some(Self::read(&real_path, name)?)),
+            None => Ok(None),
+        }
+    }
+
+    fn read(path: &PathBuf, name: &str) -> Result<String, ConfigProcessingError> {
+        fs::read_to_string(path).or_else(|err| {
+            Err(ConfigProcessingError {
+                reason: format!("Unable to read info {name:?} from {path:?}: {err}"),
+            })
+        })
+    }
+}
+
+pub struct ProcessedCSConfig {
+    pub listeners: Vec<ListenerConfig>,
+    pub infos: Infos,
+}
+
+#[derive(Debug)]
+pub struct ConfigProcessingError {
+    reason: String,
+}
+
+impl std::error::Error for ConfigProcessingError {}
```

Comment:
Prefer deriving Error over implementing it by hand - use thiserror::Error and see the other examples in the crate

Reply:
Was looking for something like that, apparently I missed it! Thanks

Reply:
and changed